### PR TITLE
GODRIVER-1924 Remove skipReason from estimatedDocumentCount test for versioned API

### PR DESCRIPTION
**Pull request by benjirewis**
[GODRIVER-1924](https://jira.mongodb.org/browse/GODRIVER-1924)

The now un-skipped test won't pass until the "latest" server version compiles on this [commit](https://evergreen.mongodb.com/version/mongodb_mongo_master_075e51fefecc54e219c56e75c3ba08c993b5c187#%230), so I won't merge until the test passes in CI.

### CODE REVIEW

**Review by iwysiu — APPROVED** [review]
lgtm when it passes

**Comment by benjirewis** [comment]
#628 needs to be merged to run versioned API testing on Ubuntu 18.04 before this skipReason can be removed.

**Review by divjotarora — APPROVED** [review]
LGTM. Before you merge, please verify that the test now runs in the expected tasks for the linked Evergreen patch.

**Comment by benjirewis** [comment]
image
Here's the new test passing instead of skipping on the linked patch! Should be all set.

**Comment by benjirewis** [comment]
image
Here's the new test passing instead of skipping on the linked patch! Should be all set.

**Pull request merged**
